woocommerce/woocommerce-ios
# Order Creation: Add status section

Part of https://github.com/woocommerce/woocommerce-ios/issues/5406.

## Description

This adds status section to create order flow. Compared to original design it doesn't have order number, since don't know it ahead of creating an order. Date is always "today", but can be injected later when we reuse it for editing. Status is selected from available site statuses to show correct title, default fallback is "pending".

Statuses sync is handled in multiple places on higher levels of app with `OrderStatusAction.retrieveOrderStatuses`. Even when there are no statuses preloaded it will fall back to `pending` and likely still work on order creation. For future improvements I added a [card](https://github.com/woocommerce/woocommerce-ios/projects/42#card-73919195) to backlog.

Edit button for now does nothing, status selection action will be added separately.

## Changes

- Adds `OrderStatusSection` view to `NewOrder` with its definition in separate file.
- Adds published var `orderStatus` to `NewOrderViewModel`, connects it to `OrderStatusSection` view.
- Adds ResultsController to get site statuses from Core Data and update initial `orderStatus` value.

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. (Tap "Create order" in the bottom sheet if that appears.)
4. Verify that status section appears, have correct layout, date, status badge and full status name (usually "Pending Payment").

*Optional*
1. Update `private var currentSiteStatuses` in `NewOrderViewModel` to return empty array to simulate missing data.
2. Build and run the app.
3. Open Order Creation and verify that status section has `pending` fallback status.

## Screenshots

regular bright|accessibility size dark
--|--
![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/144062421-15da81b1-4672-4af4-9506-19b08e10386b.png)|![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/144062400-16dd3782-dd59-4f84-afb7-eef1da5f1d0d.png)


---
